docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,806 @@
+files = ignition['storage'].get('files', []);
+files.append({'path': '/etc/hostname', 'mode': 420, 'contents': {'source': 'data:text/plain;charset=utf-8;base64,' + base64.encodestring('$MASTER_HOSTNAME').strip(), 'verification': {}}, 'filesystem': 'root'});
+ignition['storage']['files'] = files;
+json.dump(ignition, sys.stdout)" <master.ign >"master-$index-ignition.json"
+> done
+```
+
+This should create files `master-0-ignition.json`, `master-1-ignition.json` and `master-2-ignition.json`.
+
+If you look inside, you will see that they contain very little. In fact, most of the master Ignition is served by the Machine Config Server running on the bootstrap node and the masters contain only enough to know where to look for the rest.
+
+You can make your own changes here.
+
+**NOTE**: The worker nodes do not require any changes to their Ignition, but you can make your own by editing `worker.ign`.
+
+
+## Network Topology
+
+In this section we'll create all the networking pieces necessary to host the OpenShift cluster: network, subnet, router etc.
+
+
+### Network and Subnet
+
+We will use the `192.0.2.0/24` subnet range, but remove the first ten IP addresses from the allocation pool.
+
+These addresses will be used for the VRRP addresses managed by keepalived for high availability.
+
+```sh
+$ openstack network create openshift-qlvwv-network
+$ openstack subnet create --subnet-range 192.0.2.0/24 --allocation-pool start=192.0.2.10,end=192.0.2.254 --network openshift-qlvwv-network openshift-qlvwv-subnet
+```
+
+### Subnet DNS (optional)
+
+During deployment, the OpenShift nodes will need to be able to resolve public name records to download the OpenShift images and so on. They will also need to resolve the OpenStack API endpoint.
+
+The default resolvers are an often set up by the OpenStack administrator in Neutron. However, some deployments do not have default DNS servers set, meaning the servers are not able to resolve any records when they boot.
+
+If you are in this situation, you add resolvers to your Neutron subnet (`openshift-qlvwv-subnet`). These will be put into `/etc/resolv.conf` on your servers post-boot.
+
+For example, if you want to add the following nameservers: `198.51.100.86` and `198.51.100.87`, you can run this command:
+
+```sh
+$ openstack subnet --dns-nameserver 198.51.100.86 --dns-nameserver 198.51.100.87 openshift-qlvwv-subnet
+```
+
+**NOTE**: This step is optional and only necessary if you want to control the default resolvers your Nova servers will use.
+
+### Security Groups
+
+We will need two security groups: one for the master nodes (the control plane) and a separate one for the worker (compute) nodes.
+
+
+#### Master and Worker Security Groups
+
+```sh
+$ openstack security group create openshift-qlvwv-master
+$ openstack security group create openshift-qlvwv-worker
+```
+
+#### Master Security Group Rules
+
+```sh
+$ openstack security group rule create --description "ICMP" --protocol icmp openshift-qlvwv-master
```

Comment:
Having the prompt $ char makes it hard to use copy & paste. If you want to keep it there for consistency, how about using heredoc:

```sh
$ bash <<SG_END
openstack security group rule create --description "machine config server" --protocol tcp --dst-port 22623 --remote-ip 192.0.2.0/24 openshift-qlvwv-master
openstack security group rule create --description "SSH" --protocol tcp --dst-port 22 openshift-qlvwv-master
[...]
openstack security group rule create --description "VRRP" --protocol vrrp --remote-ip 192.0.2.0/24 openshift-qlvwv-master
SG_END
```

Reply:
Do you think it's okay to keep it everywhere else and just drop the prompt here? Or would you prefer I'd leave it out in every shell session?

Reply:
I've updated the SG rules and a few other things that would make copy/pasting really hard (the ignition for loops).

Let me know if you'd prefer to remove the prompt everywhere.